### ISSUE TITLE
Migrate to @types?

### ISSUE THREAD

**Issue by Tiliavir** [microsoftgraph/msgraph-typescript-typings#3]
According to [this MSDN blog post](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/) @types namespace should be used instead of typings, right? Is there a port of this type definition project?
If so, the documentation should be updated I guess ...

**Comment by dan-silver** [comment]
Thanks for the feedback @Tiliavir.  I'll look into adding @types support and post back here.

**Comment by Nasicus** [comment]
Hello I have the same problem in my project. As a workaround I added the following to `package.json`:

`    "postinstall": "copy \"./node_modules/microsoft-graph-typings/\" \"./node_modules/@types\""`

I have the following dependencies:
`   "microsoft-graph-typings": "^0.2.0"`

**Comment by offbeatful** [comment]
+1

**Comment by dan-silver** [comment]
I absolutely agree that we need to move away from using Typings.  Looking at the [publishing declaration files doc page](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html), it looks like there's two recommended ways to publish declaration files. One is publishing to DefinitelyTyped which gets mirrored to @types and the other is just using an npm package for typescript 2.0 users.


Since it looks like pull requests to DefinitelyTyped can take a while to be merged, I'm leaning towards using an npm package with just the types in it.  We can track issues with our types in this repo instead of potentially getting lost in the many PRs over there. Would that satisfy your use cases or are there additional benefits with us using @types?

**Comment by Tiliavir** [comment]
Well it says:
> There are two main ways you can publish your declaration files to npm:
> 1. bundling with your npm package, or
> 2. publishing to the @types organization on npm.

I understand the first one like it was done here: https://github.com/moment/moment - meaning the types are published with the actual code. This is the most feasible and reasonable approach in my opinion.

In your case this might not be an option, since this repository is providing types "only". In particular for a web api - hence there will never be a code base that is actually processing something. So I assume @types is the suggested solution for this particular scenario.

It is possible to use this project without typings anyway, my initial intention for this issue was just to point out, that there is a more sophisticated / newer approach on publishing types. Maybe it would as well suffice to update the documentation if a PR to DefinitelyTyped is too complicated - I have no idea, I have never tried that.

For me @types is the first place where I am looking for type definitions, so I had to search for a long time to find your repository. In addition it is nice to have all the "pure" type definition projects in an @types directory under node_modules. However, this is based on my shallow typescript experience. Maybe there are other people that can provide a more insightful perspective and better arguments on this issue.

**Comment by billba** [comment]
I use `npm install @types/foo -D` when installing types for library `foo`.

This project is a little unusual in that there is no companion library. It is a free-floating set of types associated with a service, not a library. So I'm not sure what `foo` would/should be. Given this, I tend to agree with @dan-silver that it should just be a standalone npm package.

(I empathize with @Tiliavir that the discoverability of said package will be low.)

**Comment by dan-silver** [comment]
I think for now we'll go with a standalone npm package, but we should definitely investigate if there's a way to make it more discoverable in the future.  I'm waiting on a package to be created under the @microsoft account at https://www.npmjs.com/~microsoft , so it should be published shortly after that.
